update components-permission docs (#728)
karmada-io/website: docs/administrator/security/component-permission.md

---
title: Component Permissions
---

# Permissions for Karmada Components

> This document provides a detailed explanation of the Role-Based Access Control (RBAC) permissions required for key components of Karmada. It aims to ensure secure management and compliant access to resources within the system. As a multi-cluster management framework, controlling permissions between Karmada components is essential for maintaining the security and stability of the entire system. This guide will help administrators understand and configure the RBAC permissions needed for Karmada components effectively.

Reviewer: > This document provides a detailed explanation of the Role-Based Access Control (RBAC) permissions required for key components of Karmada.

Currently the doc only includes `karmada-operator` and `karmada-agent`, will other components be included in the future?

Author: yes, Once the permissions for the other components have been sorted out, they will also be added to this document

For more detailed information about RBAC permissions, please refer to the [official RBAC documentation](https://kubernetes.io/docs/reference/access-authn-authz/rbac/).

> *Note: The resources in this document are all Karmada control plane resources.*

Reviewer: Is that true?
For instance, the `leases` required by `karmada-operator` is Kubernetes instead of the Karmada control plane.

Author: Great catch! This note should be placed in the "karmada-agent" section.

## karmada-operator

### Component Description:

The Karmada operator is a method for installing, upgrading, and deleting Karmada instances. It builds upon the basic Karmada resource and controller concepts, provides convenience to centrally manage entire lifecycle of Karmada instances in a global cluster.

### Component Permissions:

#### Cluster-Scope:

| Resource        | API Group           | Resource Names | Verbs                       | Description                                                  |
| --------------- | ------------------- | -------------- | --------------------------- | ------------------------------------------------------------ |

> | leases          | coordination.k8s.io | /              | get, create, update         | Required for leader election                                 |

Reviewer: What does the `/` mean? mean `all` or `NA`?

Author: it means `all`. Should I change `/` to `all`?

Author: `*` would be better

---
title: Component Permissions
---

# Karmada Component Permissions

> As a multi-cluster management framework, it is crucial to control the permissions of Karmada components based on the resources they access to maintain the security and stability of the entire system. The Karmada community has compiled a list of resources that key components require access to and has implemented the principle of least privilege for the relevant components using the community-maintained installation and deployment tool. This document aims to outline the list of resources that Karmada components need to access along with the reasons for accessing those resources, providing clear guidance for users customizing component permissions.

Reviewer: ```suggestion
This document provides a detailed explanation of the resources each Karmada component needs to access and the reasons for these accesses. It will help administrators understand and configure the RBAC permissions needed for Karmada components effectively, ensuring that the system operates securely and efficiently.

The [installation tools](https://karmada.io/docs/installation/) maintained by the community are designed with security in mind. These tools use Role-Based Access Control ([RBAC](https://kubernetes.io/docs/reference/access-authn-authz/rbac/)) to manage access to components, ensuring they only have access to the resources they require. By adhering to the principle of least privilege, these tools minimize potential security risks and prevent unauthorized access or actions within the system.
```

---
title: Component Permissions
---

# Karmada Component Permissions

This document provides a detailed explanation of the resources each Karmada component needs to access and the reasons for these accesses. It aims to ensure secure management and compliant access to resources within the system. As a multi-cluster management framework, controlling permissions between Karmada components is essential for maintaining the security and stability of the entire system.

Karmada uses [Role-Based Access Control (RBAC)](https://kubernetes.io/docs/reference/access-authn-authz/rbac/) to manage access to its components. Proper configuration of RBAC ensures that only authorized users and services can perform specific actions, enhancing security and operational integrity. This document will help administrators understand and configure the RBAC permissions needed for Karmada components effectively.

## karmada-operator

### Introduction:

The Karmada operator is a method for installing, upgrading, and deleting Karmada instances. It builds upon the basic Karmada resource and controller concepts, provides convenience to centrally manage entire lifecycle of Karmada instances in a global cluster.

> ### Access Permissions:
> 
> #### Resources in Cluster Where karmada-operator is deployed

Reviewer: ```suggestion
### Resources in Cluster Where karmada-operator is deployed
```
This docs focus on explaining the resources that each component accesses, not guiding people how to config permissions by RBAC

> ##### Cluster-Scope:

Reviewer: Are you going to classify the following resources by scope?
Note that the resources in the following table include both cluster-scoped and namespace-scoped.

| events          | ""                  | *              | create                      | Allows karmada-operator to record events in the Kubernetes API server |
| nodes           | ""                  | *              | list                        | To get Node IP                                               |
| pods            | ""                  | *              | list                        | For  pod health checks                                       |
| namespaces      | ""                  | *              | get                         | To get information about namespaces, and deploy resources into specific namespaces |
| secrets         | ""                  | *              | get, create, update, delete | To manage secrets which might contain sensitive data like credentials |
| services        | ""                  | *              | get, create, update, delete | To manage services to expose applications within the cluster |
| statefulsets    | ""                  | *              | get, create, update, delete | To manage StatefulSets, e.g., etcd                           |

> | deployments     | apps                | *              | get, create, update, delete | To manage Deployments, e.g., karmada-operator                |

Reviewer: ```suggestion
| deployments     | apps                | *              | get, create, update, delete | To manage Deployments, e.g., karmada-apiserver                |
```
The operator usually not manage itself.

| Non-Resource URLs | Verbs | Description |
|-------------|-------|-------------|
| /healthz | get | Used to check whether the Karmada API server is healthy. |

#### Resources in Karmada System

NONE.

## karmada-agent

### Introduction:

karmada-agent can register a specific cluster to the Karmada control plane and sync manifests from the Karmada control plane to the member cluster. In addition, it also syncs the status of member cluster and manifests to the Karmada control plane.

### Access Permissions:

#### Resources in Cluster Where karmada-agent is deployed

> ##### Cluster-Scope:

Reviewer: ditto.
